src/Inkluzitron/Modules/UserModule.cs

```diff
@@ -99,39 +99,73 @@ public Task UnsetGenderAsync()
 
         [Command("duck set")]
         [Summary("Nastaví přezdívku používanou v kachničce, aby bylo možné stáhnout prestiž za nákupy.")]
-        public Task SetKisNicknameAsync([Remainder][Name("prezdivka")] string nickname)
-            => UpdateKisNicknameAsync(nickname);
+        public async Task SetKisNicknameAsync([Remainder][Name("přezdívka")] string nickname)
+        {
+            if (await DbContext.Users.AnyAsync(o => o.KisNickname == nickname))
+            {
+                await ReplyAsync(Configuration["Kis:Messages:NonUniqueNick"]);
+                return;
+            }
 
-        [Command("duck unset")]
-        [Alias("duck clear")]
-        [Summary("Smaže přezdívku používanou v kachničce.")]
-        public Task UnsetKisNicknameAsync() => UpdateKisNicknameAsync(null);
+            await Patiently.HandleDbConcurrency(async () =>
+            {
+                var user = await DbContext.GetOrCreateUserEntityAsync(Context.User);
+                bool canSave = false;
 
-        public async Task SetGenderAsync(Gender gender)
-        {
-            var user = await DbContext.GetOrCreateUserEntityAsync(Context.User);
-            user.Gender = gender;
-            await DbContext.SaveChangesAsync();
-            await Context.Message.AddReactionAsync(ReactionSettings.Checkmark);
+                try
+                {
+                    if (string.IsNullOrEmpty(user.KisNickname))
+                    {
+                        user.KisNickname = nickname;
+                        canSave = true;
+                        return;
+                    }
+
+                    await ReplyAsync(Configuration["Kis:Messages:AlreadySet"]);
+                }
+                finally
```

Comment:
V tom try blocku nevidím žádný způsob jak by mohlo reálně dojít k vyjímce, pokud by v Configuration nechyběla definice té hlášky. Není to tu zbytečné?

Reply:
Tady nešlo o výjímku, ale o to, aby se neprovedl kus kodu. Ale teď mě napadá lepší řešení, takže to ještě upravím.